components/brave_wallet/renderer/brave_wallet_js_handler.cc

```diff
@@ -248,30 +263,48 @@ void BraveWalletJSHandler::InjectInitScript() {
 }
 
 void BraveWalletJSHandler::FireEvent(const std::string& event,
-                                     const std::string& event_args) {
+                                     base::Value event_args) {
   base::Value args = base::Value(base::Value::Type::LIST);
   args.Append(event);
-  args.Append(event_args);
+  args.Append(std::move(event_args));
   CallMethodOfObject(render_frame_->GetWebFrame(),
                      u"ethereum",
                      u"emit",
                      std::move(args));
 }
 
-void BraveWalletJSHandler::ConnectEvent(const std::string& chain_id) {
-  FireEvent(kConnectEvent, chain_id);
+void BraveWalletJSHandler::ConnectEvent() {
+  if (!EnsureConnected())
+    return;
+
+  brave_wallet_provider_->GetChainId(base::BindOnce(
+      &BraveWalletJSHandler::OnGetChainId, base::Unretained(this)));
+}
+
+void BraveWalletJSHandler::OnGetChainId(const std::string& chain_id) {
+  base::DictionaryValue event_args;
+  event_args.SetStringKey("chainId", chain_id);
+  FireEvent(kConnectEvent, std::move(event_args));
+  is_connected_ = true;
+  chain_id_ = chain_id;
 }
 
 void BraveWalletJSHandler::DisconnectEvent(const std::string& message) {
-  FireEvent(kDisconnectEvent, message);
+  // FireEvent(kDisconnectEvent, message);
 }
 
 void BraveWalletJSHandler::ChainChangedEvent(const std::string& chain_id) {
-  FireEvent(kChainChangedEvent, chain_id);
+  if (chain_id_ == chain_id)
+    return;
+
+  base::DictionaryValue event_args;
+  event_args.SetStringKey("chainId", chain_id);
+  FireEvent(kChainChangedEvent, std::move(event_args));
+  chain_id_ = chain_id;
 }
 
 void BraveWalletJSHandler::AccountsChangedEvent(const std::string& accounts) {
-  FireEvent(kAccountsChangedEvent, accounts);
+  // FireEvent(kAccountsChangedEvent, accounts);
```

Comment:
why is this commented out?

Reply:
disconnect and accounts changed events will be handled after connecting to UI